openstack/pkg/designate/provider.go

```diff
@@ -0,0 +1,145 @@
+package designate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/dns/v2/recordsets"
+	"github.com/libdns/libdns"
+)
+
+type Provider struct {
```

Comment:
The exported fields will need JSON tags, for names like `auth_open_stack` -- and the AuthOpenStack type too, use `camel_case` for names.

Reply:
added :-)